x-pack/filebeat/input/http_endpoint/handler.go

```diff
@@ -187,3 +207,20 @@ func newJSONDecoder(r io.Reader) *json.Decoder {
 	dec.UseNumber()
 	return dec
 }
+
+// getBodyReader returns a reader that decodes the specified Content-Encoding.
+func getBodyReader(r *http.Request) (body io.ReadCloser, status int, err error) {
+	switch enc := r.Header.Get(headerContentEncoding); enc {
+	case "gzip":
+		gzipReader, err := gzip.NewReader(r.Body)
```

Comment:
If this input processes a lot of concurrent requests, one thing to watch out for is high memory usage from allocating a new gzip reader for each request.

You can improve this by storing the readers in a `sync.Pool` to try to allow reusing them. You'd have to call https://pkg.go.dev/compress/gzip#Reader.Reset before using it in that case.

Might be a premature optimization for now though.

Reply:
I added a pooled gzip reader.